include/openthread/nat64.h

```diff
@@ -174,6 +312,28 @@ typedef void (*otNat64ReceiveIp4Callback)(otMessage *aMessage, void *aContext);
  */
 void otNat64SetReceiveIp4Callback(otInstance *aInstance, otNat64ReceiveIp4Callback aCallback, void *aContext);
 
+/**
+ * Gets the configured IPv4 CIDR in the NAT64 translator.
+ *
+ * Available when `OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE` is enabled.
+ *
+ * @param[in]  aInstance         A pointer to an OpenThread instance.
+ * @param[out] aCidr             A pointer to an otIp4Cidr. Where the configured CIDR will be filled.
+ *
+ */
+otError otNat64GetConfiguredCidr(otInstance *aInstance, otIp4Cidr *aCidr);
```

Comment:
What about something like `otNat64GetCidr`? What is the purpose of `Configured`?

I had the same question when looking at the CLI commands again.

Reply:
I use `nat64 configuredprefix` to distinguish it from `br nat64prefix` since the prefix in translator can be a standalone module.

Then I make two similar API for IPv4 CIDR and IPv6 prefix in nat64.

Reply:
Having `otBorderRoutingGetNat64Prefix()` and `otNat64GetConfiguredPrefix()` is confusing, especially for someone looking at the APIs for the first time. In what cases would we want these values to be different?

Reply:
I think these APIs are similar but tied to different modules and configs (one for translator one in routing manager). Is this correct?

We can consider defining one API and then documenting that it requires either `CONFIG_NAT64_TRANSLATOR_ENABLE` or 
`CONFIG_NAT64_BORDER_ROUTING_ENABLE` (would set the info on both when both are enabled).

Would this work?

Reply:
`otBorderRoutingGetNat64Prefix()` return the local generated NAT64 prefix and `otBorderRoutingGetFavoredNat64Prefix()` return the preferred NAT64 prefix in Routing Manager. They are similar to the `otBorderRoutingGetOmrPrefix()` and `otBorderRoutingGetFavoredOmrPrefix()`.

When `CONFIG_NAT64_TRANSLATOR_ENABLE` is true, `otNat64GetConfiguredCidr()` is expected to return the same value as `otBorderRoutingGetFavoredNat64Prefix()`.

Considering `otBorderRoutingGetNat64Prefix()` and `otBorderRoutingGetFavoredNat64Prefix()` are being used in CI tests, it might need a separate PR to consolidate the APIs.

Reply:
> I think these APIs are similar but tied to different modules and configs (one for translator one in routing manager). Is this correct?

Yes, these API are fetching values from different source, although the values are likely to be the same.

> 
> We can consider defining one API and then documenting that it requires either `CONFIG_NAT64_TRANSLATOR_ENABLE` or `CONFIG_NAT64_BORDER_ROUTING_ENABLE` (would set the info on both when both are enabled).
> 
> Would this work?

Something like:

```c++
otError otNat64GetPrefix(otInstance *aInstance, otIp6Prefix *aCidr) {
#if CONFIG_NAT64_BORDER_ROUTING_ENABLE
    return otBorderRoutingGetFavoredNat64Prefix(...)
#else
    return AsCoreType(aInstance).Get<Nat64::Translator>().GetIp6Prefix(AsCoreType(aPrefix));
#endif
}
```

However, `otBorderRoutingGetFavoredNat64Prefix` has a "otRoutePreference", maybe we can fill it with some invalid value when border routing is not enabled?

Reply:
Thanks guys. 

I like the model in 
- `otBorderRoutingGetNat64Prefix()` for **local** (what BR would have used it was deciding) 
- and `otBorderRoutingGetFavoredNat64Prefix()` for **favored** prefix (what it is actually being used) 

which is  similar to OMR prefix. I think make sense to keep these as they are.

For translator, 
- We already have `otNat64SetIp4Cidr()`. So may make sense to add its getter as `otNat64GetIp4Cidr()` (basically replacing/renaming `otNat64GetConfiguredCidr()` in this PR)
- Related to `otNat64GetConfiguredPrefix()` I would suggest removing it in this PR (not adding it for now)
  - If I am not mistaken, the prefix on translator is currently always set from `RoutingManager` and would be same as `otBorderRoutingGetFavoredNat64Prefix()`.
  - Seems like we  don't have a public API to set the prefix on `Translator` so may be fine to also not have a get prefix (from `Translator`).
  - We can consider adding get and set prefix (directly on Translater) later (if there is use-case to allow build config of `TRANSLATOR` feature without `NAT64_BR`)

Thoughts?

Reply:
Makes sense

Let's discuss the detailed API around NAT64 prefix later.

Reply:
@erjiaqing , can you also update the [PR description](https://github.com/openthread/openthread/pull/8058#issue-1345773321)?

Reply:
@abtink PTAL, thanks.